### PR TITLE
COV-161: create table view for patient statuses

### PR DESCRIPTION
**Pull request by Rafistan**
JIRA: https://soen-390-covid-tracker.atlassian.net/browse/COV-161?atlOrigin=eyJpIjoiNGE0ODhiMTBmNTk5NDM5OTk1YmNmMzU3Zjk5ZGIzYTUiLCJwIjoiaiJ9

This PR is taking care of the front end part of showing a table view of the patients statuses.
Each status row in the table can be view by clicking on the Eye button in the last column which will redirect you to the specific status report.

## Screenshots
image
image

### CODE REVIEW

File: client/src/views/pages/patient/StatusReports.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    if (!isDataLoaded) {
+        return <div></div>;
+    }
+    return (
+        <div>
+            <BreadCrumbsPage
+                breadCrumbTitle="Status Reports"
+                breadCrumbParent="Patient"
+                breadCrumbActive="Status Reports"
+            />
```
Comment (jason-gerard) [inline client/src/views/pages/patient/StatusReports.js]: This should be different for the doctor and health official roles, check the mocks

File: client/src/views/pages/patient/StatusReports.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            },
+        },
+    ];
+
+    useEffect(() => {
+        async function f() {
+            const statusReports = await getStatusReports(userId, token);
+            setStatusReports(statusReports);
+            setDataLoaded(true);
+            console.log(statusReports);
```
Comment (jason-gerard) [inline client/src/views/pages/patient/StatusReports.js]: Extra console log

```suggestion
```

File: client/src/views/pages/patient/StatusReports.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                return (
+                    <div className="d-flex">
+                        <UncontrolledDropdown>
+                            <DropdownToggle href={"/statuses/" + index.statusId} className="pe-1" tag="a">
+                                <Eye size={18} />
+                            </DropdownToggle>
+                        </UncontrolledDropdown>
+                    </div>
+                );
+            },
```
Comment (jason-gerard) [inline client/src/views/pages/patient/StatusReports.js]: There are a couple styling updates to make with this check the status inbox page / file, it has the same column

File: client/src/views/pages/patient/StatusReports.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            name: "Weight",
+            sortable: true,
+            minWidth: "50px",
+            selector: (row) => row.statusBody.weight,
+        },
+        {
+            name: "Temperature",
+            sortable: true,
+            minWidth: "50px",
+            selector: (row) => row.statusBody.temperature,
```
Comment (jason-gerard) [inline client/src/views/pages/patient/StatusReports.js]: This should have a unit, check status inbox page / file

File: client/src/views/pages/patient/StatusReports.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            name: "Last Updated",
+            sortable: true,
+            minWidth: "225px",
+            selector: (row) => row.createdOn,
+        },
+        {
+            name: "Weight",
+            sortable: true,
+            minWidth: "50px",
+            selector: (row) => row.statusBody.weight,
```
Comment (jason-gerard) [inline client/src/views/pages/patient/StatusReports.js]: This should have a unit, check status inbox page / file

File: client/src/views/pages/patient/StatusReports.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+            name: "ID",
+            sortable: true,
+            maxWidth: "100px",
+            selector: (row) => row.statusId,
+        },
+        {
+            name: "Last Updated",
+            sortable: true,
+            minWidth: "225px",
+            selector: (row) => row.createdOn,
```
Comment (jason-gerard) [inline client/src/views/pages/patient/StatusReports.js]: The time needs to be formatted correctly, check status inbox page / file

**Review by jason-gerard — COMMENTED** [review]
Can you add screenshots for doctor / health official and for mobile

**Comment by Rafistan** [comment]
Screen Shot 2022-03-12 at 7 54 29 PM
Screen Shot 2022-03-12 at 7 54 34 PM
Screen Shot 2022-03-12 at 7 54 40 PM

**Review by jason-gerard — APPROVED** [review]

**Pull request merged**